### ISSUE TITLE
Some CAMERA improvement ideas

### ISSUE THREAD

**Issue by eirisocherry** [ripieces/advancedfx#827]
I have no idea how hard it is to make a better system for campath curve, so there are some ideas which can make the current cubic interp thing a bit more comfy.

`mirv_campath fix` 	//makes the tick difference between campoints ~ the same, so the campath curve looks smooth (manually it takes too much brain energy especially if campoints are placed on 100k+ ticks)

1. If (current tick) < (first campoint tick) then take the first point camera properties (position, angles, fov)
2. If (current tick) > (last campoint tick) then take the last point camera properties (position, angles, fov)
Might be useful, cuz it's impossible to keep the camera at the same position using cubic interp (linear interp is not an option in most cases)
`mirv_campath lock 1`	//enables the feature
`mirv_campath lock 0`     //disables the feature (default)

P stands for point/campath point/campoint
`mirv_input p0`           //takes campoint0 (first campoint) camera properties (position, angles, fov)
`mirv_input pLast`      //takes last campoint camera properties (position, angles, fov)
`mirv_input p0+tick`   //takes campoint0 camera properties (position, angles, fov) + tick it's placed on

So basically, this command frees you up from having to write all these commands:
`mirv_input position`, `mirv_input rotation`, `mirv_input fov`, `demo_goto`.

~~Why don't switch `mirv_campath print` properties format **from**
`Current ( x , y , z ) fov ( pitch , yaw , roll )...` **to** `Current ( x y z ) fov ( pitch yaw roll )...`
or why don't make `mirv_input` work with commas, so that, for example, `mirv_input position -592.966309 , -1907.645752 , -82.423470` actually works rather than just providing a description of what it can do.~~

^ Latest HLAE version has the commas removed. //Dec

**Comment by Dechno1337** [comment]
> Why don't switch `mirv_campath print` properties format **from** `Current ( x , y , z ) fov ( pitch , yaw , roll )...` **to** `Current ( x y z ) fov ( pitch yaw roll )...`

What are your thoughts on multiple spaces in between values for easier readability? @ShyRikki 
(`mirv_input position/angles` would still work when pasting extra spaces.)

Example image of regular space:

![bild](https://github.com/advancedfx/advancedfx/assets/40070135/d0fa996e-05d3-4476-8b0e-290292372d56)

Example image of two spaces in between values:

![bild](https://github.com/advancedfx/advancedfx/assets/40070135/5ff21654-d2f4-4a1c-9679-a4cb901704d7)

Not sure if there are any use-cases where multiple spaces could cause inconvenience for someone. 🤔

**Comment by eirisocherry** [comment]
cool

**Comment by dtugend** [comment]
Hi. Thank you for your suggestion. The implementation of this feature in a reasonable time-frame is currently not realistic for me. After considering that together with issue #896 I am closing this request as not planned for now, since it would block other things I would like to still get done before that. Sorry for the copy pasta.

**Comment by eirisocherry** [comment]
you've done more than enough, thank you
